holoviews/operation/datashader.py

```diff
@@ -1935,3 +1935,71 @@ def _sort_by_distance(cls, raster, df, x, y):
         dx, dy = np.array(xs) - x, np.array(ys) - y
         distances = pd.Series(dx*dx + dy*dy)
         return df.iloc[distances.argsort().values]
+
+
+class inspect(Operation):
+    """
+    Generalized inspect operation that detects the appropriate indicator
+    type.
+    """
+
+    pixels = param.Integer(default=3, doc="""
+       Number of pixels in data space around the cursor point to search
+       for hits in. The hit within this box mask that is closest to the
+       cursor's position is displayed.""")
+
+    null_value = param.Number(default=0, doc="""
+       Value of raster which indicates no hits. For instance zero for
+       count aggregator (default) and commonly NaN for other (float)
+       aggregators. For RGBA images, the alpha channel is used which means
+       zero alpha acts as the null value.""")
+
+    value_bounds = param.NumericTuple(default=None, length=2, allow_None=True, doc="""
+       If not None, a numeric bounds for the pixel under the cursor in
+       order for hits to be computed. Useful for count aggregators where
+       a value of (1,1000) would make sure no more than a thousand
+       samples will be searched.""")
+
+    hits = param.DataFrame(default=pd.DataFrame(), allow_None=True)
+
+    max_indicators = param.Integer(default=1, doc="""
+       Maximum number of indicator elements to display within the mask
+       of size pixels. Points are prioritized by distance from the
+       cursor point. This means that the default value of one shows the
+       single closest sample to the cursor. Note that this limit is not
+       applies to the hits parameter.""")
+
+    transform = param.Callable(default=identity, doc="""
+      Function that transforms the hits dataframe before it is passed to
+      the Points element. Can be used to customize the value dimensions
+      e.g. to implement custom hover behavior.""")
+
+    # Stream values and overrides
+    streams = param.ClassSelector(default=dict(x=PointerXY.param.x,
+                                               y=PointerXY.param.y),
+                                  class_=(dict, list))
+    x = param.Number(default=0)
+    y = param.Number(default=0)
```

Comment:
Can't you inherit all this from `inspect_base`?

Reply:
There is a clash on the `_process` method but I agree that I would like to avoid this parameter duplication. Maybe `inspect` can own the parameters, `inspect_base` can inherit from it?

Reply:
Doesn't the process just get overridden anyway?

Reply:
That will work but I'll have to define this once the concrete operations exist:

```
    _dispatch = {Points : inspect_points,
                 Polygons : inspect_poly}
```

Reply:
No objection to changing the hierarchy though.

Reply:
Implemented in bbf79f094

Reply:
I guess it doesn't have to be a linear chain but this approach means `inspect` doesn't have a load of methods inherited from `inspect_base` that it doesn't need.